packages/core/configuration/configurationSlot.ts

```diff
@@ -197,10 +197,13 @@ export default function ConfigSlot(
       get expr() {
         if (self.isCallback) {
           // compile as jexl function
-          return stringToJexlExpression(
-            String(self.value),
-            getEnv(self).pluginManager?.jexl,
-          )
+          const { pluginManager } = getEnv(self)
+          if (!pluginManager && typeof jest === 'undefined') {
+            console.warn(
+              'no pluginManager detected on config env (if you dynamically instantiate a config, for example in renderProps for your display model, check that you add the env argument)',
+            )
```

Comment:
If we do a warning like this I'm worried that people will use it in this way over time and warnings will accumulate.

Should we just `throw` here? Or make the jexl context be required by `stringToJexlExpression`?

Reply:
there will be many instances in our tests especially where an env would need to be added for configSchema.create if we want to hard throw an error or make required

Reply:
if we just make it required by stringToJexlExpression it may involve a bit less issues in tests, e.g. many tests dont actually use jexl, so instead of throwing and only "requiring" (e.g. not creating one dynamically) may be a bit better